#iZ and CWS
o http://www.openoffice.org/issues/show_bug.cgi?id=40179
o http://eis.services.openoffice.org/EIS2/servlet/cws.ShowCWS?Id=2030&Path=SRC680%2Fpj16
o pj16

Description

o source/config/ldap/makefile.mk needs -lcompat since re_comp is defined here.
o FreeBSD doesn't have -ldl
o INC+= -DNP_LINUX is also requried for FreeBSD

Index: source/config/ldap/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/config/ldap/makefile.mk,v
retrieving revision 1.4
diff -u -r1.4 makefile.mk
--- extensions/source/config/ldap/makefile.mk	15 Nov 2004 14:14:22 -0000	1.4
+++ extensions/source/config/ldap/makefile.mk	8 Jan 2005 02:48:29 -0000
@@ -116,6 +116,9 @@
         $(CPPULIB) \
         $(SALHELPERLIB) \
         $(SALLIB)
+.IF "$(OS)"=="FREEBSD"
+SHL1STDLIBS+=-lcompat
+.ENDIF
 
 DEF1NAME=$(SHL1TARGET)
 DEF1EXPORTFILE=exports.dxp    
Index: source/nsplugin/source/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/nsplugin/source/makefile.mk,v
retrieving revision 1.8
diff -u -r1.8 makefile.mk
--- extensions/source/nsplugin/source/makefile.mk	3 Jan 2005 09:04:51 -0000	1.8
+++ extensions/source/nsplugin/source/makefile.mk	8 Jan 2005 02:48:30 -0000
@@ -78,7 +78,7 @@
 	@echo GTK disabled - nothing to build
 
 .ELSE           # "$(ENABLE_GTK)"==""
-.IF "$(OS)"=="LINUX"
+.IF "$(OS)"=="LINUX" || "$(OS)"=="FREEBSD"
 INC+= -DNP_LINUX
 .ENDIF
 PKGCONFIG_MODULES=gtk+-2.0
@@ -99,8 +99,10 @@
 .IF "$(GUI)"=="UNX"
 SHL1OBJS+=$(SLO)$/npunix.obj
 SLOFILES+=$(SLO)$/npunix.obj
+.IF "$(OS)"!="FREEBSD"
 SHL1STDLIBS+= -ldl -lnsl
-.ENDIF
+.ENDIF #FREEBSD
+.ENDIF #UNX
 .IF "$(GUI)"=="WNT"
 SHL1OBJS+=$(SLO)$/npwin.obj
 SLOFILES+=$(SLO)$/npwin.obj
@@ -131,7 +133,7 @@
 
 .IF "$(GUI)"=="UNX"
 APP1STDLIBS+=$(PKGCONFIG_LIBS:s/-lpangoxft-1.0//)
-.IF "$(OS)"=="LINUX"
+.IF "$(OS)"=="LINUX" || "$(OS)"=="FREEBSD"
 .ELSE
 APP1STDLIBS+= -ldl -lnsl -lnls -lsocket
 .ENDIF
